test/new-e2e/tests/agent-platform/step-by-step/step_by_step_test.go

```diff
@@ -0,0 +1,267 @@
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+	"github.com/stretchr/testify/require"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+var osVersion = flag.String("osversion", "", "os version to test")
+var platform = flag.String("platform", "", "platform to test")
+var cwsSupportedOsVersion = flag.String("cws-supported-osversion", "", "list of os where CWS is supported")
+var architecture = flag.String("arch", "", "architecture to test (x86_64, arm64))")
+var flavorName = flag.String("flavor", "datadog-agent", "package flavor to install")
+var majorVersion = flag.String("major-version", "7", "major version to test (6, 7)")
+
+type stepByStepSuite struct {
+	e2e.Suite[e2e.VMEnv]
+	osVersion    float64
+	cwsSupported bool
+}
+
+func ExecuteWithoutError(t *testing.T, client *common.TestClient, cmd string, args ...any) {
+	var finalCmd string
+	if len(args) > 0 {
+		finalCmd = fmt.Sprintf(cmd, args...)
+	} else {
+		finalCmd = cmd
+	}
+	_, err := client.VMClient.ExecuteWithError(finalCmd)
+	require.NoError(t, err)
+}
+
+func TestStepByStepScript(t *testing.T) {
+	osMapping := map[string]ec2os.Type{
+		"debian":      ec2os.DebianOS,
+		"ubuntu":      ec2os.UbuntuOS,
+		"centos":      ec2os.CentOS,
+		"amazonlinux": ec2os.AmazonLinuxOS,
+		"redhat":      ec2os.RedHatOS,
+		"windows":     ec2os.WindowsOS,
+		"fedora":      ec2os.FedoraOS,
+		"suse":        ec2os.SuseOS,
+	}
+
+	archMapping := map[string]e2eOs.Architecture{
+		"x86_64": e2eOs.AMD64Arch,
+		"arm64":  e2eOs.ARM64Arch,
+	}
+
+	platformJSON := map[string]map[string]map[string]string{}
+
+	err := json.Unmarshal(platforms.Content, &platformJSON)
+	require.NoErrorf(t, err, "failed to umarshall platform file: %v", err)
+
+	osVersions := strings.Split(*osVersion, ",")
+	cwsSupportedOsVersionList := strings.Split(*cwsSupportedOsVersion, ",")
+
+	fmt.Println("Parsed platform json file: ", platformJSON)
+	for _, osVers := range osVersions {
+		osVers := osVers
+		cwsSupported := false
+		for _, cwsSupportedOs := range cwsSupportedOsVersionList {
+			if cwsSupportedOs == osVers {
+				cwsSupported = true
+			}
+		}
+
+		t.Run(fmt.Sprintf("test step by step on %s %s", osVers, *architecture), func(tt *testing.T) {
+			tt.Parallel()
+			fmt.Printf("Testing %s", osVers)
+			slice := strings.Split(osVers, "-")
+			var version float64
+			if len(slice) == 2 {
+				version, err = strconv.ParseFloat(slice[1], 64)
+				require.NoError(tt, err)
+			} else if len(slice) == 3 {
+				version, err = strconv.ParseFloat(slice[1]+"."+slice[2], 64)
+				require.NoError(tt, err)
+			} else {
+				version = 0
+			}
+			e2e.Run(tt, &stepByStepSuite{cwsSupported: cwsSupported, osVersion: version}, e2e.EC2VMStackDef(ec2params.WithImageName(platformJSON[*platform][*architecture][osVers], archMapping[*architecture], osMapping[*platform])), params.WithStackName(fmt.Sprintf("step-by-step-test-%v-%v-%s", os.Getenv("CI_PIPELINE_ID"), osVers, *architecture)))
```

Comment:
I think you should add the agent version in the stack name. Otherwise we can have lock conflict when running in the CI

Reply:
By curiosity how so 🤔 ? AFAIK we aren't running multiple agent version on the same ci job

Reply:
Nice catch 👍